Versioning policy (#148)
reactjs/pt-BR.reactjs.org: content/docs/faq-versioning.md

---
id: faq-versioning

> title: Politica de Versionamento

Reviewer: na Navbar traduzimos para:
```suggestion
title: Política de Versão
```

permalink: docs/faq-versioning.html
layout: docs
category: FAQ
---

> React segue os principios de [versionamento semântico (semver)](https://semver.org/).

Reviewer 2: ```suggestion
O React segue os principios de [versionamento semântico (semver)](https://semver.org/).
```

> React segue os principios de [versionamento semântico (semver)](https://semver.org/).

Reviewer 3: ```suggestion
React segue os princípios de [versionamento semântico (semver)](https://semver.org/).
```

> Isso significa que com um número de versão **x.y.z**:  

Reviewer: ```suggestion
Isso significa que com um número de versão **x.y.z**:

```

> * Ao liberarmos uma **atualização que quebra compatibilidade**, fazemos uma **major release** alterando o número **x** (ex: 15.6.2 para 16.0.0).

Reviewer 3: Colocar uma linha para espaço.

Reviewer: Na verdade abaixo dessa linha não tem espaço:

Reviewer 3: Sim, era acima da linha mesmo. =)

* Ao liberarmos uma **atualização com novas funcionalidades**, fazemos uma **minor release** alterando o número **y** (ex: 15.6.2 para 15.7.0).

> * Ao liberarmos uma **atualização para correção de erros**, fazemos um **patch release** alterando o número **z** (ex: 15.6.2 para 15.6.3).

Reviewer: ```suggestion
* Ao liberarmos uma **atualização para correção de erros**, fazemos um **patch release** alterando o número **z** (ex: 15.6.2 para 15.6.3).

```

> Atualizações que quebram compatibilidade podem também conter novas funcionalidades, e qualquer versão pode incluir correção de erros.

Reviewer 3: Aqui precisa de uma linha de espaço também.

### Atualizações que quebram compatibilidade {#breaking-changes} 

Atualizações que quebram compatibilidade são inconvenientes para todos, sendo assim, tentamos diminuir o número de major releases – por exemplo, React 15 foi lançado em Abril de 2016 e React 16 foi lançado em Setembro de 2017; React 17 não é esperado até 2019.

Todavia, lançamos novas funcionalidades em versões menores. Isso significa que updates menores são mais interessantes e convenientes que versões maiores, apesar de parecer o contrário.

### Compromisso com a Estabilidade {#commitment-to-stability} 

> Conforme o React é alterado, tentamos minimizar o esforço necessário para tirar um melhor aproveitamento das novas funcionalidades. Quando possível mantemos uma versão anterior da API em funcionamento, mesmo que isso signifique colocá-la em outro pacote. Por exemplo, [mixins são desencorajados há anos](/blog/2016/07/13/mixins-considered-harmful.html) porém atualmente continuam suportados [via create-react-class](/docs/react-without-es6.html#mixins) e muitas bases de código continuam a usá-los em versões estáveis, versões de legado.

Reviewer 2: Acho que da pra melhorar um pouco a parte final aqui:
```suggestion
Conforme o React é alterado, tentamos minimizar o esforço necessário para tirar um melhor aproveitamento das novas funcionalidades. Quando possível mantemos uma versão anterior da API em funcionamento, mesmo que isso signifique colocá-la em outro pacote. Por exemplo, [mixins são desencorajados há anos](/blog/2016/07/13/mixins-considered-harmful.html) porém atualmente continuam suportados [via create-react-class](/docs/react-without-es6.html#mixins) e muitas bases de código continuam a usá-los em código legado estável.
```

> Conforme o React é alterado, tentamos minimizar o esforço necessário para tirar um melhor aproveitamento das novas funcionalidades. Quando possível mantemos uma versão anterior da API em funcionamento, mesmo que isso signifique colocá-la em outro pacote. Por exemplo, [mixins são desencorajados há anos](/blog/2016/07/13/mixins-considered-harmful.html) porém atualmente continuam suportados [via create-react-class](/docs/react-without-es6.html#mixins) e muitas bases de código continuam a usá-los em versões estáveis, versões de legado.

Reviewer 3: ```suggestion
Conforme o React é alterado, tentamos minimizar o esforço necessário para tirar um melhor proveito das novas funcionalidades. Quando possível mantemos uma versão anterior da API em funcionamento, mesmo que isso signifique colocá-la em outro pacote. Por exemplo, [mixins são desencorajados há anos](/blog/2016/07/13/mixins-considered-harmful.html) porém atualmente continuam sendo suportados [via create-react-class](/docs/react-without-es6.html#mixins) e muitas bases de código continuam a usá-los em versões estáveis e legadas.
```

> Mais de um milhão de desenvolvedores utilizam o React, coletivamente mantendo milhões de componentes. Apenas o código base do Facebook possui mais de 50.000 componentes. Isso significa que precisamos deixá-lo o mais simples possível para atualizar para novas versões do React; Se nós realizarmos grandes mudanças sem um plano de migração, pessoas ficarão presas nas versões anteriores. As atualizações são testadas no próprio Facebook – se nosso time de menos de 10 pessoas consegue atualizar 50.000+ de componentes sozinhos, nós esperamos que a atualização seja possível para qualquer um que use o React. Em muitos casos, escrevemos [testes automatizados](https://github.com/reactjs/react-codemod) afim de atualizar componentes da sintaxe, nos quais incluímos na versão de código fonte para uso de todos.

Reviewer 2: Ficou com uma confusão na parte final:
```suggestion
Mais de um milhão de desenvolvedores utilizam o React, coletivamente mantendo milhões de componentes. Apenas o código base do Facebook possui mais de 50.000 componentes. Isso significa que precisamos deixá-lo o mais simples possível para atualizar para novas versões do React; Se nós realizarmos grandes mudanças sem um plano de migração, pessoas ficarão presas nas versões anteriores. As atualizações são testadas no próprio Facebook – se nosso time de menos de 10 pessoas consegue atualizar 50.000+ de componentes sozinhos, nós esperamos que a atualização seja possível para qualquer um que use o React. Em muitos casos, escrevemos [testes automatizados](https://github.com/reactjs/react-codemod) afim de atualizar a sintaxe de componentes, os quais incluímos na versão em código aberto para uso de todos.
```

Reviewer 2: Legal a tradução! Fiz algumas pequenas sugestões.

> Mais de um milhão de desenvolvedores utilizam o React, coletivamente mantendo milhões de componentes. Apenas o código base do Facebook possui mais de 50.000 componentes. Isso significa que precisamos deixá-lo o mais simples possível para atualizar para novas versões do React; Se nós realizarmos grandes mudanças sem um plano de migração, pessoas ficarão presas nas versões anteriores. As atualizações são testadas no próprio Facebook – se nosso time de menos de 10 pessoas consegue atualizar 50.000+ de componentes sozinhos, nós esperamos que a atualização seja possível para qualquer um que use o React. Em muitos casos, escrevemos [testes automatizados](https://github.com/reactjs/react-codemod) afim de atualizar componentes da sintaxe, nos quais incluímos na versão de código fonte para uso de todos.

Reviewer 3: ```suggestion
Mais de um milhão de desenvolvedores utilizam o React, coletivamente mantendo milhões de componentes. Apenas o código base do Facebook possui mais de 50.000 componentes. Isso significa que precisamos deixá-lo o mais simples possível para atualizar para novas versões do React; Se nós realizarmos grandes mudanças sem um plano de migração, pessoas ficarão presas nas versões anteriores. As atualizações são testadas no próprio Facebook – se nosso time de menos de 10 pessoas consegue atualizar 50.000+ de componentes sozinhos, nós esperamos que a atualização seja possível para qualquer um que use o React. Em muitos casos, escrevemos [scripts automatizados](https://github.com/reactjs/react-codemod) afim de atualizar componentes da sintaxe, nos quais incluímos na versão de código fonte para uso de todos.
```

### Atualizações Graduais através de Advertências {#gradual-upgrades-via-warnings}

> Aplicações desenvolvidas em React incluem muitas advertências úteis. Sempre que possível, nós adicionamos advertências afim de preparar para grandes alterações futuras que possam quebrar a compatibilidade. Dessa forma, se a sua aplicação não possui nenhuma advertência na ultima versão, ela será compatível com a proxima grande atualização. Isso permitirá a você atualizar os componentes do seu aplicativo um a um.  

Reviewer 3: ```suggestion
Aplicações desenvolvidas em React incluem muitas advertências úteis. Sempre que possível, nós adicionamos advertências afim de preparar para grandes alterações futuras que possam quebrar a compatibilidade. Dessa forma, se a sua aplicação não possui nenhuma advertência na última versão, ela será compatível com a próxima grande atualização. Isso permitirá a você atualizar os componentes do seu aplicativo um a um.  
```

As advertências de desenvolvimento não afetarão o desempenho da sua aplicação. Dessa maneira, você pode se sentir confiante de que sua aplicação se comportará da mesma maneira entre as versões de desenvolvimento e produção – as únicas diferenças serão de que a versão de produção não apresentará logs de advertência, tornando-a mais eficiente. (Caso você note o contrário, por gentileza, abra uma issue).

### O que Conta como uma Atualização que Quebra Compatibilidade? {#what-counts-as-a-breaking-change}

No geral, *não* alteramos o número de major version por alterações como:

> * **Advertências de Desenvolvimento.** Já que estas não afetam o comportamento de produção, nós podemos adicionar novas advertências ou modificar as existentes entre novas versões. De fato, isso é o que nos permite garantir a segurança das proximas versões. 

Reviewer 3: ```suggestion
* **Advertências de Desenvolvimento.** Já que estas não afetam o comportamento de produção, nós podemos adicionar novas advertências ou modificar as existentes entre novas versões. De fato, isso é o que nos permite garantir a segurança das próximas versões. 
```

* **APIs començando com `unstable_`.** Estas são providas como funcionalidades experimentais das quais ainda não possuímos certa estabilidade das APIs. Ao liberarmos estas com um prefixo `unstable_`, podemos rapidamente iterar e tornar a API estável de forma mais rápida.  

> * **Versões Alpha e canary do React.** Nós provemos versões alpha do React como uma maneira de testar novas funcionalidades de forma mais rápida, porém precisamos da flexibilidade para realizar alterações baseadas no que aprendemos com o período da versão alpha. Caso você faça uso destas versões, note que as APIs podem mudar antes do lançamento da versão estável. 

Reviewer 3: ```suggestion
* **Versões alpha e canary do React.** Nós provemos versões alpha do React como uma maneira de testar novas funcionalidades de forma mais rápida, porém precisamos da flexibilidade para realizar alterações baseadas no que aprendemos com o período da versão alpha. Caso você faça uso destas versões, note que as APIs podem mudar antes do lançamento da versão estável. 
```

* **APIs não documentadas e estrutura de dados interna.** Se você acessa propriedades com nomes internos como `__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED` ou `__reactInternalInstance$uk43rzhitjg`, não há garantia alguma. Você está por conta própria.

> Este documento é destinado a ser pragmático: obviamente, não queremos causar dores de cabeça a você. Se nós alterarmos a major version para todas essas mudanças, nós terminariamos liberando mais major versions e consequentemente causando mais dores de versionamento para a comunidade. Isso também significaria que não poderiamos melhorar o React tão rápido quanto gostaríamos.

Reviewer 3: ```suggestion
Este documento é destinado a ser pragmático: obviamente, não queremos causar dores de cabeça a você. Se nós alterarmos a major version para todas essas mudanças, nós terminaríamos liberando mais major versions e consequentemente causando mais dores de versionamento para a comunidade. Isso também significaria que não poderíamos melhorar o React tão rápido quanto gostaríamos.
```

### Atualizações que quebram compatibilidade {#breaking-changes} 

Atualizações que quebram compatibilidade são inconvenientes para todos, sendo assim, tentamos diminuir o número de major releases – por exemplo, React 15 foi lançado em Abril de 2016 e React 16 foi lançado em Setembro de 2017; React 17 não é esperado até 2019.

Todavia, lançamos novas funcionalidades em versões menores. Isso significa que updates menores são mais interessantes e convenientes que versões maiores, apesar de parecer o contrário.

### Compromisso com a Estabilidade {#commitment-to-stability} 

Conforme o React é alterado, tentamos minimizar o esforço necessário para tirar um melhor proveito das novas funcionalidades. Quando possível mantemos uma versão anterior da API em funcionamento, mesmo que isso signifique colocá-la em outro pacote. Por exemplo, [mixins são desencorajados há anos](/blog/2016/07/13/mixins-considered-harmful.html) porém atualmente continuam sendo suportados [via create-react-class](/docs/react-without-es6.html#mixins) e muitas bases de código continuam a usá-los em versões estáveis e legadas.

> Mais de um milhão de desenvolvedores utilizam o React, coletivamente mantendo milhões de componentes. Apenas o código base do Facebook possui mais de 50.000 componentes. Isso significa que precisamos deixá-lo o mais simples possível para atualizar para novas versões; Se nós realizarmos grandes mudanças sem um plano de migração, as pessoas poderão ficar presas em versões anteriores. As atualizações são testadas no próprio Facebook – se nosso time de menos de 10 pessoas consegue atualizar 50.000+ de componentes sozinhos, nós esperamos que a atualização seja possível para qualquer um que use o React. Em muitos casos, escrevemos [testes automatizados](https://github.com/reactjs/react-codemod) afim de atualizar componentes da sintaxe, nos quais incluímos na versão de código fonte para uso de todos.

Reviewer 3: `codemods` não são testes automatizados, são scripts.

```suggestion
Mais de um milhão de desenvolvedores utilizam o React, coletivamente mantendo milhões de componentes. Apenas o código base do Facebook possui mais de 50.000 componentes. Isso significa que precisamos deixá-lo o mais simples possível para atualizar para novas versões; Se nós realizarmos grandes mudanças sem um plano de migração, as pessoas poderão ficar presas em versões anteriores. As atualizações são testadas no próprio Facebook – se nosso time de menos de 10 pessoas consegue atualizar 50.000+ de componentes sozinhos, nós esperamos que a atualização seja possível para qualquer um que use o React. Em muitos casos, escrevemos [scripts automatizados](https://github.com/reactjs/react-codemod) afim de atualizar componentes da sintaxe, nos quais incluímos na versão de código fonte para uso de todos.
```

Reviewer 3: Atualizar a sintaxe dos componentes...

---
id: faq-versioning
title: Política de Versão
permalink: docs/faq-versioning.html
layout: docs
category: FAQ
---

O React segue os princípios de [versionamento semântico (semver)](https://semver.org/).

Isso significa que com um número de versão **x.y.z**:  

* Ao liberarmos uma **atualização que quebra compatibilidade**, fazemos uma **major release** alterando o número **x** (ex: 15.6.2 para 16.0.0).

* Ao liberarmos uma **atualização com novas funcionalidades**, fazemos uma **minor release** alterando o número **y** (ex: 15.6.2 para 15.7.0).

> 

Reviewer: aqui também não, use a versão em ingles como base
```suggestion
```

Author: Atualizada conforme versão em inglês!
